### DISCUSSION TITLE
A setting to make markdown by default

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7684] [General]
To make md the default syntax, we neet to make viewtemplate buttons like `new here – Create a new tiddler tagged with this one` also produce markdown tiddler.

So there need to be a setting in control panel for setting Default mime type.

Should this be a plugin or in the core? I think for this setting it can be in the core.
and there can be a plugin that transform text field of action-createTiddler from wikitext to md, so all user-land plugins automatically become md plugin. (or maybe not necessarily, these plugin can provide a md version, or use list widget to switch to md mode, by reading the Default mime type in the preference.

**Comment by Jermolene** [comment]
Hi @linonetwo I think it makes sense to add this to the core. We already have the translatable strings for the content types, and so should be able to show a dropdown in control panel allowing the default tiddler type to be specified. The setting should affect buttons like new tiddler, new journal, new here, etc. and also the type of tiddlers created as a result of clicking on a missing link.

**Comment by pmario** [comment]
>I think it makes sense to add this to the core.

I think it should be part of the markdown-plugin. IMO we should prepare the core in a way that it will be easy to add such settings and make it future-proof for other markup plugins too. 

I did play with the "missing link" in the tiddler body some time ago. .. **But** the approach was not really good. See PR draft: **Missing tiddler cascade - To extend the "missing tiddler hint" in the missing tiddler body** #6364

**Reply by Jermolene** [reply]
The setting to default to Markdown tiddlers could reside in the Markdown plugin, but the ability to do that relies on a new underlying mechanism that enables new tiddlers to be given a custom type; that will certainly require a core modification.
